### ISSUE TITLE
y shape and y_pred shape not same 

### ISSUE THREAD

**Issue by imrankhan441** [tensorflow/tensorflow#40406]
why those two are different 

![tfp](https://user-images.githubusercontent.com/26671669/84497611-f3b32380-acd0-11ea-9538-6e1f60f32c19.png)

**Comment by amahendrakar** [comment]
@imrankhan441,
In order to expedite the trouble-shooting process, could you please provide the complete code to reproduce the issue reported here and also the TensorFlow version you are using. Thanks!

**Comment by imrankhan441** [comment]
`from tensorflow.distribute import get_strategy
from tensorflow.distribute.experimental import TPUStrategy
from tensorflow.config import experimental_connect_to_cluster
from tensorflow.tpu.experimental import initialize_tpu_system
from tensorflow.distribute.cluster_resolver import TPUClusterResolver


from tensorflow.keras.layers import LSTM
from tensorflow.keras.layers import Dense
from tensorflow.keras.layers import Embedding
from tensorflow.keras.models import Sequential


from tensorflow.keras.datasets import imdb

number_of_words = 20000
max_len = 100

(X_train, y_train), (X_test, y_test) = imdb.load_data(num_words = number_of_words)

from tensorflow.keras.preprocessing.sequence import pad_sequences

X_train = pad_sequences(X_train, maxlen = max_len)
X_test = pad_sequences(X_test, maxlen = max_len)



try:
    tpu = TPUClusterResolver()
    print('Running on TPU ', tpu.master())
except:
    tpu = None

if tpu:
    experimental_connect_to_cluster(tpu)
    initialize_tpu_system(tpu)
    strategy = TPUStrategy(tpu)

else:
    strategy = get_strategy()

with strategy.scope():
    embedding_rnn = Sequential()
    embedding_rnn.add(Embedding(number_of_words, 10, input_shape = (X_train.shape[1], )))
    embedding_rnn.add(LSTM(units = 10, return_sequences = True, activation = "tanh"))
    embedding_rnn.add(LSTM(units = 20, return_sequences = True, activation = "tanh"))
    embedding_rnn.add(LSTM(units = 30, return_sequences = True, activation = "tanh"))
    embedding_rnn.add(LSTM(units = 40, activation = "tanh"))

    embedding_rnn.add(Dense(units = 1, activation = "sigmoid", name = "output_layer"))
    embedding_rnn.compile(optimizer = "adam", loss = "binary_crossentropy", metrics = ["accuracy"])
    
embedding_rnn.fit(X_train, y_train, batch_size = 20 * strategy.num_replicas_in_sync, epochs = 50)

y_pred = embedding_rnn.predict_classes(X_test)
y_pred.shape

`

**Comment by amahendrakar** [comment]
Was able to reproduce the issue, please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/f06bac5b647b95b26f9e6d6614bbe8dc/40406.ipynb#scrollTo=yxeH6LVBVNvL). Thanks!

**Comment by jhseu** [comment]
This looks similar to a bug that @tf-marissaw is looking at.

**Comment by tf-marissaw** [comment]
I can reproduce this using the above colab but I cannot reproduce this bug on the most recent internal version of TensorFlow. It looks like its root cause was already fixed. The bug I am looking into appears to be unrelated to this.
